extensions/ql-vscode/test/vscode-tests/no-workspace/data-extensions-editor/model-details/model-details-data-provider.test.ts

```diff
@@ -89,4 +95,40 @@ describe("ModelDetailsDataProvider", () => {
       expect(onDidChangeTreeDataListener).toHaveBeenCalledTimes(1);
     });
   });
+
+  describe("getChildren", () => {
+    const externalApiUsages: ExternalApiUsage[] = [
+      mockedObject<ExternalApiUsage>({ supported: true }),
```

Comment:
I see that the desire here is to avoid the pain of providing full objects, but `mockedObject` is going to potentially cause some issues here that will be hard to debug for future developers. I think you've already hit this, as evidenced by needing to pass `usages: undefined` in some cases.

In `ModelDetailsDataProvider` we use `isExternalApiUsage` to determine if the object we have is an `ExternalApiUsage` or a `Usage`. Unfortunately this check won't work correctly when given an object created using `mockedObject`, or at least you have to know exactly what you're doing and pass the right field values to make `isExternalApiUsage` behave correctly. While it is possible to have everything work correctly like this, it's quite error prone.

While it's is great for things like `CodeQLCliServer`, I would recommend not using `mockedObject` for types like `ExternalApiUsage` and `Usage`. How much effort would it be to not do this?